Automattic/woocommerce-subscriptions-core
# Use try/catch when calling process_payment to avoid fatals.

Fixes #330, please see for details.

Please also see discussion in p1670014386600359-slack-C029ME0FYRZ for how this affects processing subscription payment method changes with WooCommerce Payments in particular.

## Description

See #330 for a description.

## How to test this PR

Use the Change Payment Method instructions:

https://woocommerce.com/document/subscriptions/customers-view/#section-11

To simulate a failure, you could either block the network request w/ Developer Tools or use a test card that will decline.
Instead of a fatal error because of a failure, you should instead see a notice with the error.

## Product impact
